chains/evm/voter/voter.go

```diff
@@ -156,7 +157,27 @@ func (v *EVMVoter) shouldVoteForProposal(prop *proposal.Proposal, tries int) (bo
 		return v.shouldVoteForProposal(prop, tries)
 	}
 
-	return true, nil
+	return v.simulateCall(prop, 0)
+}
+
+func (v *EVMVoter) simulateCall(prop *proposal.Proposal, tries int) (bool, error) {
+	Sleep(time.Duration(rand.Intn(shouldVoteCheckPeriod)) * time.Second)
+	input, err := calls.PrepareVoteProposalInput(prop.Source, prop.DepositNonce, prop.ResourceId, prop.Data)
```

Comment:
Probably just create SimulateVoteProposal in calls package and move this logic there and here just call that